### ISSUE TITLE
OutOfMemoryException when pushing very large package

### ISSUE THREAD

**Issue by wldevries** [Inedo/upack#41]
**Version**
upack 2.2.2.2

**Description**
I'm trying to push a `upack` file of 1.5 GB and the `upack.exe` tool crashes almost immediately with an `OutOfMemoryException`.

```
PS D:\upack> upack push .\VideoCourse.fr-FR.upack http://orange:8624/upack/Content/ --user=Admin:Admin
Package: SilverFit/VideoCourse.fr-FR
Version: 1.0.0
```

**Comment by apapadimoulis** [comment]
@BenLubar I suspect there is a way to solve the OOM through better streaming, but we'll still run into the "infamous 2GB limit". How about we add this optional argument and new behavior?

chunkSize - for files exceeding this size (in megabytes), the package will be transfered using the chunked transfer API; specify 0 to disable chunking completely

I guess this change would mandate that we go to version 2.3?

**Comment by BenLubar** [comment]
@u5surf I'm pretty sure feeding the `*os.File` through an `io.Pipe` won't affect memory usage but it will use up some extra CPU cycles.

**Comment by gdivis** [comment]
@wldevries This should be fixed as of v2.2.3, at least in the .NET version. (See #46)
